### PR TITLE
Feature[CX-1699]: Palette Input V3

### PR DESCRIPTION
**Pull request by AlicanAkyuz**
The type of this PR is: **Feature**

This PR resolves [CX-1699]

### Description

- Implements specs for Palette V3 Input component 
- replaces inputs in the codebase with the new V3 Input – except the ones in BiddingAddress and PriceRangeOptions (they will be refactored in another PR)

#### IOS
<div>

</ div>

#### Android
<div>

</ div>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes
Inputs are replaced in all platforms

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1699]: https://artsyproduct.atlassian.net/browse/CX-1699

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Inputs are replaced in all platforms)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5997077b04e79def8d494f90f7723c9b3f5633ae
</p>

File: src/palette/elements/Input/Input.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { storiesOf } from "@storybook/react-native"
+import SearchIcon from "lib/Icons/SearchIcon"
+import { Flex, Input } from "palette"
+import React from "react"
+import { View } from "react-native"
+import { withTheme } from "storybook/decorators"
+import { List } from "storybook/helpers"
+
+storiesOf("InputV3", module)
```
Comment (pvinis) [inline src/palette/elements/Input/Input.stories.tsx:9]: nice!

**Review by pvinis — APPROVED** [review]

**Comment by pvinis** [comment]
could you rerun the secrets thing here? 🙏

**Review by olerichter00 — APPROVED** [review]
Great work  👐

**Comment by AlicanAkyuz** [comment]
@pvinis do you know the command to update .secrets.baseline?

**Comment by AlicanAkyuz** [comment]
I've run that one but it creates no diff locally. may I just resolve the conflict here?

**Comment by pvinis** [comment]
@olerichter00 help 😅

File: src/palette/elements/Input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,330 @@
+import _ from "lodash"
+import { Color, EyeOpenedIcon, Flex, Sans, Spinner, ThemeV3Type, useTheme, XCircleIcon } from "palette"
```
Comment (pvinis) [inline src/palette/elements/Input/Input.tsx:2]:
```suggestion
import { Color, EyeOpenedIcon, Flex, Sans, Spinner, useTheme, XCircleIcon } from "palette"
```

File: src/palette/elements/Input/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+    const localClear = () => {
+      input.current?.clear()
+      localOnChangeText("")
+      rest.onClear?.()
+    }
+
+    useImperativeHandle(ref, () => input.current!)
+
+    const fontFamily = (theme as ThemeV3Type).fonts.sans.regular
```
Comment (pvinis) [inline src/palette/elements/Input/Input.tsx:98]:
```suggestion
    const fontFamily = theme.fonts.sans.regular
```

File: src/palette/elements/Input/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+      // otherwise just return the shortest placeholder
+      return placeholder[placeholder.length - 1]
+    }
+
+    return (
+      <Flex flexGrow={1} style={containerStyle}>
+        <InputTitle required={required}>{title}</InputTitle>
+
+        {!!description && (
+          <Sans color="black60" size="2" style={{ marginBottom: 5 }}>
```
Comment (MounirDhahri) [inline src/palette/elements/Input/Input.tsx:198]: [already solved]: I apologize I didn't get to review this earlier. We moved from using `Sans` for text and we are now exclusively using `Text` from palette instead.

**Pull request merged**